server/server.go

```diff
@@ -80,10 +80,8 @@ func (h *serverAuthProxyHandler) handler(w http.ResponseWriter, r *http.Request)
 	// before we handle any requests, make sure the user is authenticated
 	if h.credentials.Username != "" || h.credentials.Password != "" {
 		u, p, ok := r.BasicAuth()
-		if !ok {
+		if !ok || h.credentials.Username != u || h.credentials.Password != p {
```

Comment:
is this assuming that `!ok` always means StatusUnauthorized? couldn't it be for instance 500 err?

Reply:
Interesting. HTTP spec named 401 wrong :) Unauthorized but should be unauthenticated. Nice catch. http://www.dirv.me/blog/2011/07/18/understanding-403-forbidden/index.html 

As for @abonas  question, I believe !ok will happen if something is wrong with the basic auth header (maybe it isn't there, or maybe the base64 encoding is bad or maybe there is no basic auth in the request). I don't think this kind of thing would make us return a 500 - we should still return a 401 which is simply "you have not successfully authenticated with the server". I imply this via the  Go docs on this function: https://golang.org/pkg/net/http/#Request.BasicAuth

I'm OK to merge this.